Sources/GRPC/AsyncAwaitSupport/GRPCAsyncResponseStreamWriter.swift

```diff
@@ -17,28 +17,116 @@
 #if compiler(>=5.6)
 
 /// Writer for server-streaming RPC handlers to provide responses.
+///
+/// To enable testability this type provides a static ``GRPCAsyncResponseStreamWriter/makeResponseStreamWriter()``
+/// method which allows you to create a stream that you can drive.
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public struct GRPCAsyncResponseStreamWriter<Response: Sendable>: Sendable {
+  /// An `AsyncSequence` backing a ``GRPCAsyncResponseStreamWriter`` for testing purposes.
+  ///
+  /// - Important: This `AsyncSequence` is never finishing.
+  public struct ResponseStream: AsyncSequence {
+    public typealias Element = (Response, Compression)
+
+    @usableFromInline
+    internal let stream: AsyncStream<(Response, Compression)>
+
+    @inlinable
+    init(stream: AsyncStream<(Response, Compression)>) {
+      self.stream = stream
+    }
+
+    public func makeAsyncIterator() -> AsyncIterator {
+      AsyncIterator(iterator: self.stream.makeAsyncIterator())
+    }
+
+    public struct AsyncIterator: AsyncIteratorProtocol {
+      @usableFromInline
+      internal var iterator: AsyncStream<(Response, Compression)>.AsyncIterator
+
+      @inlinable
+      init(iterator: AsyncStream<(Response, Compression)>.AsyncIterator) {
+        self.iterator = iterator
+      }
+
+      public mutating func next() async -> Element? {
+        await self.iterator.next()
+      }
+    }
+  }
+
+  /// Simple struct for the return type of ``GRPCAsyncResponseStreamWriter/makeResponseStreamWriter()``.
+  ///
+  /// This struct contains two properties:
+  /// 1. The ``writer`` which is the actual ``GRPCAsyncResponseStreamWriter`` and should be passed to the method under testing.
+  /// 2. The ``stream`` which can be used to observe the written responses.
+  public struct ResponseStreamWriter {
```

Comment:
Can we make the naming a little clearer that it's for testing? The names on the request side are 👌

Reply:
Yeah slightly awkward naming these types. I ended up with `TestingStreamWriter` and `makeTestingResponseStreamWriter()` which aligns with the request side